cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -209,142 +176,161 @@ pub mod pallet {
+		/// further messages from the channel.
 		///
 		/// - `origin`: Must pass `Root`.
 		/// - `new`: Desired value for `QueueConfigData.drop_threshold`
 		#[pallet::call_index(4)]
 		#[pallet::weight((T::WeightInfo::set_config_with_u32(),DispatchClass::Operational,))]
 		pub fn update_drop_threshold(origin: OriginFor<T>, new: u32) -> DispatchResult {
 			ensure_root(origin)?;
-			QueueConfig::<T>::mutate(|data| data.drop_threshold = new);
 
-			Ok(())
+			QueueConfig::<T>::try_mutate(|data| {
+				data.drop_threshold = new;
+				data.validate::<T>()
+			})
 		}
 
-		/// Overwrites the number of pages of messages which the queue must be reduced to before it
-		/// signals that message sending may recommence after it has been suspended.
+		/// Overwrites the number of pages which the queue must be reduced to before it signals
+		/// that message sending may recommence after it has been suspended.
 		///
 		/// - `origin`: Must pass `Root`.
 		/// - `new`: Desired value for `QueueConfigData.resume_threshold`
 		#[pallet::call_index(5)]
 		#[pallet::weight((T::WeightInfo::set_config_with_u32(), DispatchClass::Operational,))]
 		pub fn update_resume_threshold(origin: OriginFor<T>, new: u32) -> DispatchResult {
 			ensure_root(origin)?;
-			QueueConfig::<T>::mutate(|data| data.resume_threshold = new);
 
-			Ok(())
+			QueueConfig::<T>::try_mutate(|data| {
+				data.resume_threshold = new;
+				data.validate::<T>()
+			})
 		}
+	}
 
-		/// Overwrites the amount of remaining weight under which we stop processing messages.
-		///
-		/// - `origin`: Must pass `Root`.
-		/// - `new`: Desired value for `QueueConfigData.threshold_weight`
-		#[pallet::call_index(6)]
-		#[pallet::weight((T::WeightInfo::set_config_with_weight(), DispatchClass::Operational,))]
-		pub fn update_threshold_weight(origin: OriginFor<T>, new: Weight) -> DispatchResult {
-			ensure_root(origin)?;
-			QueueConfig::<T>::mutate(|data| data.threshold_weight = new);
-
-			Ok(())
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn integrity_test() {
+			let w = Self::on_idle_weight();
+			assert!(w != Weight::zero());
+			assert!(w.all_lte(T::BlockWeights::get().max_block));
 		}
 
-		/// Overwrites the speed to which the available weight approaches the maximum weight.
-		/// A lower number results in a faster progression. A value of 1 makes the entire weight
-		/// available initially.
-		///
-		/// - `origin`: Must pass `Root`.
-		/// - `new`: Desired value for `QueueConfigData.weight_restrict_decay`.
-		#[pallet::call_index(7)]
-		#[pallet::weight((T::WeightInfo::set_config_with_weight(), DispatchClass::Operational,))]
-		pub fn update_weight_restrict_decay(origin: OriginFor<T>, new: Weight) -> DispatchResult {
-			ensure_root(origin)?;
-			QueueConfig::<T>::mutate(|data| data.weight_restrict_decay = new);
+		fn on_idle(_block: BlockNumberFor<T>, limit: Weight) -> Weight {
```

Comment:
Could we move all this code from `on_idle` into a function in `migrations.rs` ? Seems like a sort of lazy migration.

Reply:
yep done